File: solutions/go/0070_爬楼梯.go
package main

import "fmt"

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	a, b := 1, 2
	for i := 3; i <= n; i++ {
		a, b = b, a+b
	}
	return b
}

func main() {
	cases := [][2]int{{1, 1}, {2, 2}, {3, 3}, {5, 8}, {10, 89}, {45, 1836311903}}
	for _, tc := range cases {
		result := climbStairs(tc[0])
		ok := "PASS"
		if result != tc[1] {
			ok = "FAIL"
		}
		fmt.Printf("climbStairs(%d) = %d, expected %d: %s\n", tc[0], result, tc[1], ok)
	}
}

File: solutions/java/0070_爬楼梯.java
package com.leetcode;

public class Solution {
    public int climbStairs(int n) {
        if (n <= 2) return n;
        int a = 1, b = 2;
        for (int i = 3; i <= n; i++) {
            int tmp = a + b;
            a = b;
            b = tmp;
        }
        return b;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[][] cases = {{1, 1}, {2, 2}, {3, 3}, {5, 8}, {10, 89}, {45, 1836311903}};
        for (int[] tc : cases) {
            int result = sol.climbStairs(tc[0]);
            String ok = result == tc[1] ? "PASS" : "FAIL";
            System.out.printf("climbStairs(%d) = %d, expected %d: %s%n", tc[0], result, tc[1], ok);
        }
    }
}
